xdsl/tools/xdsl-opt

```diff
@@ -15,6 +17,43 @@ class OptMain(xDSLOptMain):
     def register_all_arguments(self, arg_parser: argparse.ArgumentParser):
         super().register_all_arguments(arg_parser)
 
+        #arg_parser.add_argument(
+        #    "--check-coverage",
+        #    default=False,
+        #    action='store_true',
+        #    help="Generate the xDSL code coverage for this run.")
```

Comment:
Currently, the coverage is generated by default. This is a choice I made to keep the impact (and therefore the review time) as low as possible in this PR.
What we can also do is:
- only generate coverage if a flag is present (this will require adding this flag to all filecheck tests)
- generate by default, but not generate if if a flag is present
- always generate the coverage (the current version)

Furthermore, notice that these changes are in the `tools` directory and not in the big `xDSLOptMain` class. This is another choice we have. We can either leave them here, therefore not making users include this flag when building their own `...-opt` tools, or we can move it to the big class. Both are fine for me.